### PR TITLE
Musark 897

### PR DESCRIPTION
**Pull request by rituvesh**

### CODE REVIEW

File: src/models/analysis.js [hunk 1]
```diff
@@ -2,26 +2,98 @@ import {simpleGet, simplePost} from '../shared/RxAjax';
...
 
 class Analysis {
   constructor(props) {
     entries(props).forEach(([k, v]) => this[k] = v);
   }
 }
 
+Analysis.getAnalysisTypesForCollection = (ajaxGet = simpleGet) => ({museumId, collectionId, token, callback}) => {
+  const url = Config.magasin.urls.api.analysisType.getAnalysisTypesForCollection(museumId, collectionId);
+  const call = ajaxGet(url, token, callback).map(({response}) => response);
```
Comment (jarlah) [inline src/models/analysis.js]: remove unnecessary variable. return directly.

File: src/models/analysis.js [hunk 2]
```diff
@@ -2,26 +2,98 @@ import {simpleGet, simplePost} from '../shared/RxAjax';
...
+  return call;
+};
+
+
```
Comment (jarlah) [inline src/models/analysis.js]: extra blank line

File: src/models/analysis.js [hunk 3]
```diff
@@ -2,26 +2,98 @@ import {simpleGet, simplePost} from '../shared/RxAjax';
...
+};
+
+Analysis.getAnalysisById = (ajaxGet = simpleGet) => ({museumId, id, token, callback}) => {
+  const url = `${Config.magasin.urls.api.analysis.getAnalysisById(museumId, id)}`;
+  return ajaxGet(url, token, callback)
+    .map((analysis) => analysis.response);
+};
+
+Analysis.getAnalysisTypeById = (ajaxGet = simpleGet) => ({museumId, id, token, callback}) => {
+  return ajaxGet(`${Config.magasin.urls.api.analysisType.getAnalysisTypeById(museumId, id)}`, token, callback)
```
Comment (jarlah) [inline src/models/analysis.js]: i think that we should maybe follow convention and extract the url to variable like in the methods above.

File: src/models/analysis.js [hunk 4]
```diff
@@ -2,26 +2,98 @@ import {simpleGet, simplePost} from '../shared/RxAjax';
...
+        museumId: props.museumId,
+        collectionId: props.collectionId,
+        token: props.token
+      }).map(object => {
+        if (!object) {
+          return analysis;
+        }
+        return {...analysis, museumNo: object.museumNo, subNo: object.subNo, term: object.term};
+      });
+
```
Comment (jarlah) [inline src/models/analysis.js]: remove blank line

File: src/modules/analysis/AnalysisViewComponent.js [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+    } else {
+      val = store.analysis[field];
+    }
+  } else {
+    val = '';
+  }
+  return val;
+};
+
+
```
Comment (jarlah) [inline src/modules/analysis/AnalysisViewComponent.js]: remove extra two blank lines

File: src/modules/app/AppComponent.js [hunk 6]
```diff
@@ -69,6 +70,11 @@ export class AppComponent extends Component {
     this.props.clearNodePicklist();
     const localAppSession = this.props.appSession.copy({museumId, collectionId});
     this.props.goTo(Config.magasin.urls.client.storagefacility.goToRoot(localAppSession));
+    this.props.getAnalysisTypesForCollection({
```
Comment (jarlah) [inline src/modules/app/AppComponent.js]: remove this

File: src/modules/app/AppComponent.js [hunk 7]
```diff
@@ -131,11 +137,10 @@ export class AppComponent extends Component {
               <LinkContainer to={Config.magasin.urls.client.magasin.goToMagasin(this.props.appSession)}>
                 <NavItem>{ I18n.t('musit.texts.magazine') }</NavItem>
               </LinkContainer>
-              {this.props.enableAnalysis &&
```
Comment (jarlah) [inline src/modules/app/AppComponent.js]: add this feature toggle back

File: src/modules/app/AppComponent.js [hunk 8]
```diff
@@ -197,12 +201,12 @@ const commands = {
...
   setCollectionId$,
   clearObjectPicklist$,
-  clearNodePicklist$
+  clearNodePicklist$,
+  getAnalysisTypesForCollection$
 };
 
 const props = {
-  goTo: hashHistory.push,
-  enableAnalysis: Config.isDev
```
Comment (jarlah) [inline src/modules/app/AppComponent.js]: add back

File: src/modules/app/appSession.js [hunk 9]
```diff
@@ -136,11 +135,14 @@ const loadAppSession = (ajaxGet = simpleGet, accessToken) => {
   );
 };
 
+
```
Comment (jarlah) [inline src/modules/app/appSession.js]: remove extra blank line

Comment (jarlah) [reply src/modules/app/appSession.js]: i hate extra white lines ... :/

**Review by jarlah — CHANGES_REQUESTED** [review]
Just some small minor stuff.

And, test coverage has fallen below 58%. this is unacceptable. Must be 58% or more. see test coverage report and add tests for model functions.  You can actually make simple marble tests for the model functions separately. Or integrate them into the store test. But I think doing the last maybe is overkill and time consuming.

File: src/modules/analysis/AnalysisViewComponent.js [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+    </div>
+  );
+}
+function NewLine() {
+  return <Form horizontal><FormGroup />
+    <hr/>
+  </Form>;
+}
+const getAnalysisTypeTerm = (store) => {
+  let val = '';
```
Comment (ingarabr) [inline src/modules/analysis/AnalysisViewComponent.js]: I prefer to not have a mutable variable like this but return out when we have found the value. In this case we will have two return point, from `foundType.name` and where we have the return statement today with an empty value.
I'm an lazy developer so this comment goes for the other functions in this file that follow the same pattern :)

File: src/models/analysis.js [hunk 11]
```diff
@@ -2,26 +2,94 @@ import {simpleGet, simplePost} from '../shared/RxAjax';
...
+          });
+        })).map(arrayOfObjectDetails => {
+          const actualValues = arrayOfObjectDetails.filter(a => a);
+          if (actualValues.length === 0) {
+            return analysis;
+          }
+          return {
+            ...analysis, events: analysis.events.map(e => {
+              const od = arrayOfObjectDetails.find(objD => objD.uuid === e.objectId);
+              if (!od) {
```
Comment (ingarabr) [inline src/models/analysis.js]: will this be better if we inline it?

`!od ? e : {...e, ...od}`

File: src/models/analysis.js [hunk 12]
```diff
@@ -2,26 +2,94 @@ import {simpleGet, simplePost} from '../shared/RxAjax';
...
+Analysis.getAnalysisTypeById = (ajaxGet = simpleGet) => ({museumId, id, token, callback}) => {
+  return ajaxGet(Config.magasin.urls.api.analysisType.getAnalysisTypeById(museumId, id), token, callback)
+    .map(({response}) => response);
+};
+
+Analysis.getAnalysisWithDeatils = (ajaxGet = simpleGet) => (props) => {
+  return Analysis.getAnalysisById(ajaxGet)(props)
+    .flatMap(analysis =>
+      MusitActor.getActor(ajaxGet)({token: props.token, actorId: analysis.registeredBy})
+        .map(actor => {
```
Comment (ingarabr) [inline src/models/analysis.js]: how about changing this to something like this?
`   .map(actor => (actor ?  {...analysis, registeredByName: actor.fn} : analysis))`

**Review by jarlah — APPROVED** [review]

**Pull request merged**